snuba/consumer.py

```diff
@@ -69,11 +62,7 @@ def flush_batch(self, batch):
                 replacements.append(data)
 
         if inserts:
-            write_rows(
-                self.clickhouse,
-                self.__dataset,
-                inserts
-            )
+            self.__writer.write(self.__dataset.get_schema(), inserts)
```

Comment:
long term idea: 
I think right now the dataset has a bit of a mixed responsibility: it holds the DDL, it does message processing, it can help on queries, which is ok as long as one dataset is one table.
It seems we are going towards moving more complex dataset related logic into snuba:
- joins
- eventstore query optimizations
- multi-dataset to consume from one topic and store in different tables.

What if we turned the dataset into a facade that returned a subset of the following depending on which type of dataset this is:
- a consumer class to ingest and write data
- a query component to which we delegate the query (that can perform optimizations and so on)
- a DDL component that we can use to bootstrap the table in different environments

The dataset class becomes just a thin layer that provide access to these components and does not hold any other responsibility.

What do you think ?